src/Interfaces/interest/Interest.ts

```diff
@@ -0,0 +1,20 @@
+import Code from '../code/Code';
+import User from '../user/User';
+
+export default interface Interest {
+    id: string;
+    createdAt: Date | string;
+    updatedAt: Date | string;
+    userId: number;
+    code?: string;
```

Comment:
And for these, maybe these optional relationships could be set in another interface below, and interface Interest extends relationships interface as partial, it seems this is possible with type aliases:
https://stackoverflow.com/questions/44101534/typescript-optionally-extending-interface